### PR TITLE
[WIP] 4.1 changelog

### PR DESCRIPTION
**Pull request by jennyf19**
Not everything is here yet, so WIP, but feel free to commit directly or suggest additions/changes. Will be linking up to the blog post as it gets complete.

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+
+Bug Fixes:
+- **When using the `ConfidentialClientApplicationOptions` and including, for example `Instance = "https://login.microsoftonline.com/"`, MSAL.NET was concatenating the double-slash**. MSAL.NET will now check for a trailing slash and remove it. There is no action needed on the part of the developer. See [issue 1196 for details](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1196).
+- **When using ADFS 2019, if no login-hint was included in the call, a null ref was thrown**. See [issue 1214 for details](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1214)
+- **For certain older auth libraries, there is an issue with null handlnig in json**. The json serializer in MSAL.NET no longer writes values to json for which the values are null, this is especially important for foci_id. See [issue 1189](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1189) and [issue https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1176)
```
Comment (jmprieur) [inline changelog.txt]: **On iOS,** for certain older auth libraries **sharing the cache with MSAL.NET**, there was an issue with null **handling** in json

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+- **MSAL.NET now supports ClientAssertions**. See [issue 1188](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1188)
+
+Bug Fixes:
+- **When using the `ConfidentialClientApplicationOptions` and including, for example `Instance = "https://login.microsoftonline.com/"`, MSAL.NET was concatenating the double-slash**. MSAL.NET will now check for a trailing slash and remove it. There is no action needed on the part of the developer. See [issue 1196 for details](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1196).
```
Comment (jmprieur) [inline changelog.txt]: "on the part"?
from you?

File: changelog.txt (hunk 1 above)
Comment (jennyf19) [reply changelog.txt]: @jmprieur yes, of course! nice catch...thank you.

File: changelog.txt [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+- **MSAL.NET now supports ClientAssertions**. See [issue 1188](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1188)
+- `MsalUiRequiredException` now exposes a new property named `Classification` and new string constants with values for this property, to enable you to provide the best user experience in your application when you receive an `MsalError.InvalidGrantError` when calling `AcquireTokenSilent`. See [#1148](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1148)
```
Comment (jmprieur) [inline changelog.txt]: @jennyf19 : I've added this today

File: changelog.txt [hunk 4]
```diff
@@ -1,3 +1,15 @@
+4.1.0
+=============
+New Features:
+- **MSAL.NET now provides options to control the system web browser**. From MSAL.NET 4.0.0, you have been able to use the interactive token acquisition with .NET Core, by delegating the sign-in and consent part to the system web browser on your machine. MSAL.NET 4.1, brings improvements to this experience by helping you run a specific browser if you wish, and by giving you ways to decide what to display to the user in case of a successful authentication, and in case of failure. [More information about this feature here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/msal-net-4.1#improved-experience-with-the-system-web-browser-on-net-core)
+- **MSAL.NET now supports ClientAssertions**. See [issue 1188](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1188)
```
Comment (MarkZuber) [inline changelog.txt]: nit: inconsistency on [] for issues.  This one it's [issue 1188] and the other one it's [#1148]

Comment (MarkZuber) [reply changelog.txt]: I think you can just do [1148] and github will auto-link it.

Comment (jennyf19) [reply changelog.txt]: thanks...we used to do it like that but when the repos merged, we had to specify the url..will get back to doing it this way again. thx@

---
In reply to: [298357591](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1238#discussion_r298357591) [](ancestors = 298357591)

File: changelog.txt (hunk 2 above)
Comment (jennyf19) [reply changelog.txt]: i think it's fine

---
In reply to: [298079432](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1238#discussion_r298079432) [](ancestors = 298079432)

File: changelog.txt (hunk 1 above)
Comment (jennyf19) [reply changelog.txt]: i also updated the blog post w/this bug section

---
In reply to: [298079162](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1238#discussion_r298079162) [](ancestors = 298079162)

**Pull request merged**